app/forms/idv/inherited_proofing/base_form.rb

```diff
@@ -0,0 +1,149 @@
+module Idv
+  module InheritedProofing
+    class BaseForm
+      include ActiveModel::Model
+
+      class << self
+        def model_name
+          ActiveModel::Name.new(self, nil, namespaced_model_name)
+        end
+
+        def namespaced_model_name
+          self.to_s.gsub('::', '')
+        end
+
+        def fields
+          @fields ||= required_fields + optional_fields
+        end
+
+        def required_fields
+          raise NotImplementedError,
+                'Override this method and return an Array of required field names as Symbols'
+        end
+
+        def optional_fields
+          raise NotImplementedError,
+                'Override this method and return an Array of optional field names as Symbols'
+        end
+      end
+
+      private_class_method :namespaced_model_name, :required_fields, :optional_fields
+
+      attr_reader :payload_hash
+
+      validate :validate_field_names
+
+      def initialize(payload_hash:)
+        raise ArgumentError, 'payload_hash is blank?' if payload_hash.blank?
+        raise ArgumentError, 'payload_hash is not a Hash' unless payload_hash.is_a? Hash
+
+        self.class.attr_accessor(*self.class.fields)
+
+        @payload_hash = payload_hash.dup
+
+        populate_field_data
+      end
+
+      def submit
+        validate
+
+        FormResponse.new(
+          success: valid?,
+          errors: errors,
+          extra: {
+          },
+        )
+      end
+
+      # Perhaps overkill, but a mapper service of some kind, not bound to this class,
+      # that takes into consideration context, may be more suitable. In the meantime,
+      # simply return a Hash suitable to place into flow_session[:pii_from_user] in
+      # our inherited proofing flow steps.
+      def user_pii
+        raise NotImplementedError, 'Override this method and return a user PII Hash'
+      end
+
+      private
+
+      attr_writer :payload_hash
+
+      # Populates our field data from the payload hash.
+      def populate_field_data
+        payload_field_info.each do |field_name, field_info|
+          # Ignore fields we're not interested in.
+          next unless respond_to? field_name
+
+          value = payload_hash.dig(
+            *[field_info[:namespace],
+              field_info[:field_name]].flatten.compact,
+          )
+          public_send("#{field_name}=", value)
+        end
+      end
+
+      # Validator for field names. All fields (not the presence of data) are required.
+      def validate_field_names
```

Comment:
Is see the comment here, but if `payload_fields` is a hash I'm not sure I see how "field is missing" and "data is missing" are meaningfully different.

Reply:
This was one of the original requirements - to make sure all the _fields themselves_ are present in the payload, not 100% sure just why; however, it will save some time troubleshooting if we fail to receive the fields we're expecting, rather a "<Field name> must be filled in/blank/etc." red herring. TBH, it causes some headaches, but alas the "requirements".

Reply:
Where is this a requirement? It does seem like a superfluous validation

Reply:
@gangelo - This requirement doesn't make sense, let's get rid of it.

@alexanderocarson - Let's revisit this to see what we might want to do here in the future, perhaps?

Reply:
Great ty.

Reply:
Done! @benjaminchait @jmhooper